core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -571,7 +571,8 @@ private[deploy] class Master(
       usableWorkers: Array[WorkerInfo],
       spreadOutApps: Boolean): Array[Int] = {
     val coresPerExecutor = app.desc.coresPerExecutor
-    val minCoresPerExecutor = coresPerExecutor.getOrElse(1)
+    val coresPerTask = app.desc.coresPerTask
+    val minCoresPerExecutor = math.max(coresPerExecutor.getOrElse(1), coresPerTask)
```

Comment:
I don't think we should do this. If the user explicitly specified `spark.executor.cores` they expect their executors to be scheduled with that many cores, no more no less.